ui/src/widgets/ui-file-input/UIFileInput.vue

```diff
@@ -0,0 +1,161 @@
+            :label="label" :prepend-icon="icon"
+            :accept="accept"
+            :multiple="multiple"
+            variant="outlined"
+            hide-details="auto"
+            :rules="[maxFileSize]"
+        />
+        <div v-else-if="!uploaded" class="nrdb-ui-file-input--progress">
+            <v-progress-linear :model-value="progress" :height="24">
+                <template #default="{ value }">
+                    <strong>{{ Math.ceil(value) }}%</strong>
+                </template>
+            </v-progress-linear>
+        </div>
+        <div v-else class="nrdb-ui-file-input--uploaded">
+            <v-icon icon="mdi-check-circle" color="success" />
+            <label class="v-label">File Uploaded!</label>
+            <label class="v-label" @click="reset()">
+                <a class="nrdb-anchor">Upload Another File</a>
+            </label>
+        </div>
+        <v-btn variant="flat" :disabled="!files || uploading || uploaded || maxFileSize(files)" @click="upload(files)">
+            {{ uploading ? 'Uploading...' : 'Upload' }}
+        </v-btn>
+    </div>
+</template>
+
+<script>
+
+import { useDataTracker } from '../data-tracker.mjs' // eslint-disable-line import/order
+import { mapState } from 'vuex' // eslint-disable-line import/order
+
+export default {
+    name: 'DBUIFileInput',
+    inject: ['$socket'],
+    props: {
+        id: { type: String, required: true },
+        props: { type: Object, default: () => ({}) },
+        state: { type: Object, default: () => ({}) }
+    },
+    setup (props) {
+        useDataTracker(props.id)
+    },
+    data () {
+        return {
+            files: null,
+            uploading: false,
+            uploaded: false,
+            progress: 0
+        }
+    },
+    computed: {
+        ...mapState('data', ['messages']),
+        icon: function () {
+            const icon = this.props.icon
+            return icon ? 'mdi-' + icon.replace(/^mdi-/, '') : null
+        },
+        accept: function () {
+            return this.props.accept
+        },
+        label: function () {
+            return this.props.label
+        },
+        showFileSize: function () {
+            return this.props.showFileSize
+        },
+        multiple: function () {
+            return this.props.multiple
+        }
+    },
+    methods: {
+        formatFileSize (bytes) {
+            if (bytes === 0) return '0 Bytes'
+            const k = 1000
+            const dm = 2
+            const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+            const i = Math.floor(Math.log(bytes) / Math.log(k))
+            return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i]
+        },
+        reset: function () {
+            this.files = null
+            this.uploading = false
+            this.uploaded = false
+            this.progress = 0
+        },
+        upload: function (file) {
+            if (file && !this.multiple) {
```

Comment:
I cannot see how multiple files are uploaded Joe?

Reply:
yep - good point, I got caught in two minds about `multiple`, and appears I forgot i was halfway with it - will address

Reply:
Addressed in https://github.com/FlowFuse/node-red-dashboard/pull/981/commits/1fe18a05881f60fd743016f375281a02e8f3cb81